lib/Db/TagMapper.php

```diff
@@ -145,20 +145,22 @@ public function getAllTagsForMessages(array $messages): array {
 
 		$qb = $this->db->getQueryBuilder();
 		$idsQuery = $qb->select('mt.*')
+			->addSelect('t.*')
 			->from('mail_message_tags', 'mt')
+			->join('mt', $this->getTableName(), 't', $qb->expr()->eq('mt.tag_id','t.id'))
 			->where(
-				$qb->expr()->in('imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
+				$qb->expr()->in('mt.imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
 			);
 		$idsQuery = $idsQuery->execute();
 		$queryResult = $idsQuery->fetchAll();
 		if (empty($queryResult)) {
 			return [];
 		}
-		$result = [];
-		foreach ($queryResult as $qr) {
-			$result[] = $qr['imap_message_id'];
-			$result[$qr['imap_message_id']][] = $this->getTag((int)$qr['tag_id']);
-		};
+		$result = array_map(function ($qr) {
+			$transformed[] = $qr['imap_message_id'];
+			$transformed[$qr['imap_message_id']] = $qr;
+			return $transformed;
+		}, $queryResult);
 		return $result;
```

Comment:
this method is supposed to return a Tag[][] but as far as I can see it's a generic array[][] built from row arrays

Reply:
Yeah the join doesn't return the entity apperntly. I don't know enough about the query builder to change that. Any suggestions?

Reply:
if you pass the select query into `\OCP\AppFramework\Db\QBMapper::findEntities` you should get an array of entities back. you'll then have to group them per message.

Reply:
won't work because I don't have a mapper for the message_tags

Reply:
that's actually why I didn't use a join in the first place, because I need the enttity to further proceed in the code.

Reply:
you were right, we can't use `findEntities` directly but we can still build the entities in a loop if we omit the columns that are not meant for the entitiy. see my latest changes :)